client/src/pages/AddFood.js

```diff
@@ -1,37 +1,122 @@
-import React, { useState, useEffect } from "react";
+import React, { useState } from "react";
+import { useMutation } from "react-query";
 import axios from "axios";
-import Card from "@mui/material/Card";
-import Fab from "@mui/material/Fab";
-import AddIcon from "@mui/icons-material/Add";
+import Button from "@mui/material/Button";
 import Layout from "../components/Layout";
+import styled from "styled-components";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
 
 function AddFood() {
   const [food, setFood] = useState("");
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm();
+  const onSubmit = (data) => console.log(data);
+  console.log(watch("example"));
   const url = "http://localhost:5000/food";
 
-  useEffect(() => {
-    axios
-      .post(url, { title: "title", img: "img", author: "author" })
-      .then((response) => {
-        setFood(response.data);
-      });
-  }, [url]);
+  const mutation = useMutation({
+    mutationFn: (newFood) => {
+      return axios
+        .post(url, { title: "title", img: "img", author: "author" })
```

Comment:
this line can be changed to:

```suggestion
        .post(url, newFood)
```